activesupport/lib/active_support/file_evented_update_checker.rb

```diff
@@ -0,0 +1,72 @@
+require 'listen'
+
+module ActiveSupport
+  class FileEventedUpdateChecker
+    attr_reader :listener
+    def initialize(files, directories={}, &block)
+      @files = Set.new
```

Comment:
I see... Well, the next line will override it anyway. If I understand it correctly, you could remove this one and use `to_set` on the next line. Unless you have no plans to change `@files` later then you could use `uniq!`.